### PR TITLE
Fixing issue with compilation diagnostics

### PR DESCRIPTION
**Pull request by fabiocav**
Fixing issue when getting compilation and analyzer diagnostics. 
The previous version was not correctly getting compilation diagnostics in the script diagnostics check pass.

### CODE REVIEW

**Comment by fabiocav** [comment]
@davidebbo, @mathewc: found this issue when testing the latest deployed bits. I'll have some tests for this but wanted to get the fix in ASAP.

**Comment by mathewc** [comment]
Yes, please add tests, but I'm fine with you getting this in first if you're confident on it.

:ship:

File: src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilation.cs [hunk 1]
```diff
@@ -29,7 +29,8 @@ public CSharpCompilation(Compilation compilation)
 
         public ImmutableArray<Diagnostic> GetDiagnostics()
         {
-            return _compilation.WithAnalyzers(GetAnalyzers()).GetAllDiagnosticsAsync().Result;
+            var diagnostics = _compilation.WithAnalyzers(GetAnalyzers()).GetAllDiagnosticsAsync().Result;
```
Comment (mathewc) [inline src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilation.cs]: Its confusing that you have to do to calls here - why is that? GetAllDiagnostics should, well, get them ALL right?

**Comment by mathewc** [comment]
Merged

File: src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilation.cs (hunk 1 above)
Comment (fabiocav) [inline src/WebJobs.Script/Description/DotNet/CSharp/CSharpCompilation.cs]: GetAllDiagnostics gets all the analyzer diagnostics and errors. It's just poorly named.

**Pull request closed without merging**